provisionersdk/archive.go

```diff
@@ -38,17 +51,27 @@ func dirHasExt(dir string, exts ...string) (bool, error) {
 }
 
 func DirHasLockfile(dir string) (bool, error) {
-	return dirHasExt(dir, ".terraform.lock.hcl")
+	return DirHasLockfileFS(osFS, dir)
+}
+
+func DirHasLockfileFS(afs afero.Fs, dir string) (bool, error) {
+	return dirHasExt(afs, dir, ".terraform.lock.hcl")
 }
 
 // Tar archives a Terraform directory.
 func Tar(w io.Writer, logger slog.Logger, directory string, limit int64) error {
+	return TarFS(osFS, w, logger, directory, limit)
+}
+
+// TarFS archives a Terraform directory to afs.
+// The filesystem **must** support reading symbolic links.
```

Comment:
Security-related question: do we need to support symbolic links? Perhaps we should check if the attacker is able to submit a link to `/etc/something` and read it via Coder UI.